stable/artifactory/values.yaml

```diff
@@ -429,6 +429,9 @@ artifactory:
         url: "jdbc:postgresql://{{ .Release.Name }}-postgresql:{{ .Values.postgresql.service.port }}/{{ .Values.postgresql.postgresqlDatabase }}"
         driver: org.postgresql.Driver
         username: "{{ .Values.postgresql.postgresqlUsername }}"
+        {{- if .Values.postgresql.postgresqlPassword }}
+        password: "{{ .Values.postgresql.postgresqlPassword }}"
```

Comment:
@woodcockjosh [This](https://github.com/helm/charts/blob/master/stable/postgresql/templates/secrets.yaml#L16) creates a secret for `.Values.postgresql.postgresqlPassword` from postgresql sub-chart and in statefulset it is referenced using **JF_SHARED_DATABASE_PASSWORD**
```
{{ if or .Values.database.secrets.password .Values.database.password .Values.postgresql.enabled }}
        - name: JF_SHARED_DATABASE_PASSWORD
          valueFrom:
            secretKeyRef:
        {{- if .Values.database.secrets.password }}
              name: {{ tpl .Values.database.secrets.password.name . }}
              key: {{ tpl .Values.database.secrets.password.key . }}
        {{- else if .Values.database.password }}
              name: {{ template "artifactory.fullname" . }}-database-creds
              key: db-password
        {{- else if .Values.postgresql.enabled }}
              name: {{ .Release.Name }}-postgresql
              key: postgresql-password
        {{- end }}
      {{- end }}
```

Reply:
right well until a solution is implemented for #859 I'm going to use this approach to fix the problem with upgrades. It would be a shame if everyone else had to copy paste the artifactory chart to solve their problem but I understand if upgrading the chart without the upgrade failing is not the highest priority for the repo.